### PR TITLE
[ PIP-82] [pulsar-broker] Fix a couple of issues

### PR DESCRIPTION
**Pull request by bharanic-dev**
### Modifications

- Handle ChildrenChanged metadata-store event.
- Update the publish rate limiter correctly.

### Verifying this change

- [x] Make sure that the change passes the CI checks.
- [x] verified via manual testing

### Documentation

No documentation impact.

### CODE REVIEW

**Comment by bharanic-dev** [comment]
@jerrypeng PTAL when you get a chance.

**Comment by ravi-vaidyanathan** [comment]
LGTM

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/resourcegroup/ResourceGroupPublishLimiter.java [hunk 1]
```diff
@@ -160,12 +168,22 @@ private void replaceLimiters(Runnable updater) {
 
     @Override
     public void close() {
-      this.apply();
-      replaceLimiters(null);
+        replaceLimiters(null);
```
Comment (cckellogg) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/resourcegroup/ResourceGroupPublishLimiter.java:171]: Does the order of calls matter here? If so maybe add a comment?

Comment (bharanic-dev) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/resourcegroup/ResourceGroupPublishLimiter.java]: @cckellogg thank you. I added the comment. I also fixed the case where the callback was not getting called when ratelimiter is closed (not invoking the callback can leave the producer blocked forever). PTAL.

**Review by cckellogg — APPROVED** [review]

**Pull request merged**